compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/CodeGeneratorImpl.kt

```diff
@@ -96,6 +98,19 @@ class CodeGeneratorImpl(
         associate(sources, path)
     }
 
+    override fun associateWithClasses(
+        classes: List<KSClassDeclaration>,
+        packageName: String,
+        fileName: String,
+        extensionName: String
+    ) {
+        val path = pathOf(packageName, fileName, extensionName)
+        val files = classes.map {
+            it.containingFile ?: NoSourceFile(projectBase, it.qualifiedName?.asString().toString())
```

Comment:
should the `NoSourceFile` be cached based on qualified name?

Reply:
It's only used to calculate the path name and will be GC'd.